### ISSUE TITLE
language/disassemble_test hits assert

### ISSUE THREAD

**Issue by jensjoha** [dart-lang/sdk#28790]
Test was initially ignored here: https://github.com/dart-lang/sdk/commit/f95701bf6119f41dfcda921268ff07836bd86898
Was removed (by mistake I suppose) here: https://github.com/dart-lang/sdk/commit/fc2458b8423979573ef3bdc70edf9711b38a99b2

But I couldn't fint an issue for it.

```
(lots of output)
tderr:
../../runtime/vm/object.h: 3557: error: expected: kind() != RawScript::kKernelTag
Dumping native stack trace for thread 6487
  [0x00000000009a1ee3] dart::Profiler::DumpStackTrace()
  [0x00000000009a1ee3] dart::Profiler::DumpStackTrace()
  [0x000000000068f0a1] dart::DynamicAssertionHelper::Fail(char const*, ...)
  [0x000000000091dea4] dart::Parser::Parser(dart::Script const&, dart::ParsedFunction*, dart::TokenPosition)
  [0x000000000092228d] dart::Parser::ParseFunction(dart::ParsedFunction*)
  [0x00000000006f3a6b] dart::Compiler::ComputeLocalVarDescriptors(dart::Code const&)
  [0x00000000008c4450] dart::Code::GetLocalVarDescriptors() const
  [0x0000000000725790] dart::Disassembler::DisassembleCodeHelper(char const*, dart::Code const&, bool)
  [0x00000000007266e2] dart::Disassembler::DisassembleCode(dart::Function const&, bool)
  [0x00000000006f2a59] Unknown symbol
  [0x00000000006f1f3e] dart::Compiler::CompileFunction(dart::Thread*, dart::Function const&)
  [0x00000000006ed04b] dart::DRT_CompileFunction(dart::NativeArguments)
  [0x00007fbf320b8630] Unknown symbol
  [0x00007fbf320b86b0] Unknown symbol
  [0x00007fbf2ae413fb] Unknown symbol
  [0x00007fbf2ae40b74] Unknown symbol
  [0x00007fbf2ae3fd3a] Unknown symbol
  [0x00007fbf2ae3b9c9] Unknown symbol
  [0x00007fbf2ae39389] Unknown symbol
  [0x00007fbf2ae38c60] Unknown symbol
  [0x00007fbf2ae3866f] Unknown symbol
  [0x00007fbf320b8a4e] Unknown symbol
  [0x000000000070ac28] dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)
-- End of DumpStackTrace


Command[vm]: DART_CONFIGURATION=DebugX64 out/DebugX64/dart --dfe=utils/kernel-service/kernel-service.dart --disassemble --print-variable-descriptors --no-background-compilation --ignore-unrecognized-flags --packages=/b/build/slave/vm-kernel-linux-debug-x64-be/build/sdk/.packages /b/build/slave/vm-kernel-linux-debug-x64-be/build/sdk/tests/language/disassemble_test.dart
Took 0:00:22.856170

Short reproduction command (experimental):
    python tools/test.py -cdartk -t120 language/disassemble_test
```

Ignoring test for now (https://codereview.chromium.org/2702553002/).

**Comment by kmillikin** [comment]
This happens when we try to compute local variable descriptors for a function that doesn't have them yet.  If the function has a pointer to a Kernel function (which might actually be a Kernel field) we will use the Kernel scope analysis on it, otherwise we will try to parse it with the parser.

This doesn't work for functions that have neither a pointer to a Kernel function nor an associated token stream, we'll end up in the parser without a token stream.  An example of such functions are invoke field dispatchers which are generated by the Kernel pipeline.

We need to either attach something to these as a Kernel function and properly interpret it, or else use a more sophisticated test (perhaps based on the script kind and the function kind) to see which pipeline to use and handle them correctly in the Kernel scope pipeline.

**Comment by nex3** [comment]
@kmillikin I'm working on getting more gardening issues assigned. If you're not the right person to assign this to, please reassign to that person.

**Comment by jensjoha** [comment]
This no longer seems to be an issue. Bisecting finds that it was fixed with 2109cc53bf58eed6c71c7fde4af5f4a334c8d3f7.
I'll create a CL to update the status file.
